readthedocs/config/config.py

```diff
@@ -55,28 +56,12 @@
 INVALID_KEYS_COMBINATION = 'invalid-keys-combination'
 INVALID_KEY = 'invalid-key'
 
-DOCKER_DEFAULT_IMAGE = 'readthedocs/build'
-DOCKER_DEFAULT_VERSION = '2.0'
+DOCKER_DEFAULT_IMAGE = getattr(settings, 'DOCKER_DEFAULT_IMAGE', 'readthedocs/build')
+DOCKER_DEFAULT_VERSION = getattr(settings, 'DOCKER_DEFAULT_VERSION', '2.0')
 # These map to corresponding settings in the .org,
 # so they haven't been renamed.
-DOCKER_IMAGE = '{}:{}'.format(DOCKER_DEFAULT_IMAGE, DOCKER_DEFAULT_VERSION)
-DOCKER_IMAGE_SETTINGS = {
-    'readthedocs/build:1.0': {
-        'python': {'supported_versions': [2, 2.7, 3, 3.4]},
-    },
-    'readthedocs/build:2.0': {
-        'python': {'supported_versions': [2, 2.7, 3, 3.5]},
-    },
-    'readthedocs/build:3.0': {
-        'python': {'supported_versions': [2, 2.7, 3, 3.3, 3.4, 3.5, 3.6]},
-    },
-    'readthedocs/build:stable': {
-        'python': {'supported_versions': [2, 2.7, 3, 3.3, 3.4, 3.5, 3.6]},
-    },
-    'readthedocs/build:latest': {
-        'python': {'supported_versions': [2, 2.7, 3, 3.3, 3.4, 3.5, 3.6]},
-    },
-}
+DOCKER_IMAGE = getattr(settings, 'DOCKER_IMAGE', 'readthedocs/build:2.0')
+DOCKER_IMAGE_SETTINGS = getattr(settings, 'CONFIG_DOCKER_IMAGE_SETTINGS', {})
```

Comment:
Why `CONFIG_` was prepended here?

I suppose it just should be `DOCKER_IMAGE_SETTINGS` as we are already using in our settings.

Reply:
I didn't get you.
I prepend the `CONFIG_` to make these settings specific to the config.py file.
However, I have pushed the changes, but using `DOCKER_IMAGE_SETTINGS` as the name is causing one test to fail.

Reply:
We have been using `DOCKER_IMAGE_SETTINGS` so we should keep using its name.

Regarding the test, please take a look to see if you can fix it or find the reason why it's failing.

Reply:
Earlier, `DOCKER_IMAGE_SETTINGS` were not defined in `settings/base.py`.

**Reason for test failure:**
since `DOCKER_IMAGE_SETTINGS` is now defined in `settings/base.py`, these lines

https://github.com/rtfd/readthedocs.org/blob/f4d3a93a5c487351e836d51066faf40511a0b45a/readthedocs/doc_builder/config.py#L59-L62

add some **more** items to the dictionary and hence the `Actual Call` and `Expected Call` are not same.

This can be fixed pretty easily in two ways:
1. Mocking `DOCKER_IMAGE_SETTINGS` to be empty dictionary for this test.
2. Adding the missing settings in the `Expected Call`.

I am +1 on first option and -0 on the second and would like to know that what is a better option?

Reply:
If the test is not about docker images and python versions, mocking as empty dict is the way to go. On the other, if we want to be sure about some specific value on that dictionary, we should mock it with that specific value.

Changing the expected call with the values of that setting is tricky, because then we will add another docker image and the test will fail again.